lib/net/http_fetch_plugin.js

```diff
@@ -69,20 +108,25 @@ shaka.net.HttpFetchPlugin = class {
     // the AbortController.
     const timeoutMs = request.retryParameters.timeout;
     if (timeoutMs) {
-      const timer = new shaka.util.Timer(() => {
+      timeoutTimer = new shaka.util.Timer(() => {
         abortStatus.timedOut = true;
         controller.abort();
       });
 
-      timer.tickAfter(timeoutMs / 1000);
-
-      // To avoid calling |abort| on the network request after it finished, we
-      // will stop the timer when the requests resolves/rejects.
-      op.finally(() => {
-        timer.stop();
-      });
+      timeoutTimer.tickAfter(timeoutMs / 1000);
     }
 
+    // To avoid calling |abort| on the network request after it finished, we
+    // will stop the timers when the requests resolves/rejects.
+    op.finally(() => {
+      if (timeoutTimer) {
+        timeoutTimer.stop();
+      }
+      if (stallTimer) {
+        stallTimer.stop();
+      }
+    });
```

Comment:
should the connectionTimer be stopped as well?

Reply:
Fixed!